catalog/cataloger_diff.go

```diff
@@ -130,24 +133,25 @@ func (c *cataloger) Diff(ctx context.Context, repository string, leftReference s
 	return differences, hasMore, nil
 }
 
-// doDiff internal implementation of the actual diff. limit <0 will scan the complete branch
-func (c *cataloger) doDiff(ctx context.Context, tx db.Tx, params *doDiffParams) error {
+func (c *cataloger) newDiffScanner(tx db.Tx, params *doDiffParams) (*diffScanner, error) {
+	var scanner *diffScanner
+	var err error
 	if params == nil {
-		return fmt.Errorf("doDiff params are required: %w", ErrInvalidValue)
+		return nil, fmt.Errorf("doDiff params are required: %w", ErrInvalidValue)
```

Comment:
True, same comment I had - prefer not to use pointers in this case unless it is a big structure, but this one will not blow our stack and as far as I know, there is no guaranty it will copy (performance) but we should assume we got a copy.

Reply:
OK